File: LeetCode/01448. Count Good Nodes in Binary Tree/01448.c
void helper(struct TreeNode *root, int max, int *count) {
    if (root != NULL) {
        if (max <= root->val) {
            max = root->val;
            ++(*count);
        }
        
        helper(root->left, max, count);
        helper(root->right, max, count);
    }
}

int goodNodes(struct TreeNode *root) {
    int count = 0;
    
    helper(root, INT_MIN, &count);
    
    return count;
}

File: LeetCode/01448. Count Good Nodes in Binary Tree/01448.go
// https://leetcode.com/problems/count-good-nodes-in-binary-tree

func goodNodes(root *TreeNode) int {
    var dfs func(*TreeNode, int) int
    dfs = func(node *TreeNode, maxVal int) int {
        if node == nil {
            return 0
        }

        count := 0
        if maxVal <= node.Val {
            maxVal = node.Val
            count++
        }

        return count + dfs(node.Left, maxVal) + dfs(node.Right, maxVal)
    }

    return dfs(root, root.Val)
}
